workers/miflora.py

```diff
@@ -20,8 +18,47 @@ def _setup(self):
       _LOGGER.debug("Adding %s device '%s' (%s)", repr(self), name, mac)
       self.devices[name] = MiFloraPoller(mac, BluepyBackend)
 
+  def config(self):
+    ret = []
+    for name, poller in self.devices.items():
+      ret += self.config_device(name)
+    return ret
+
+  def config_device(self, name):
+    ret = []
+    device={"identifiers": self.format_topic(name, separator="_"),
+            "manufacturer": "Xiaomi",
+            "model": "MiFlora",
+            "name": self.format_topic(name, separator=" ").title()}
+    for attr in monitoredAttrs:
+      if attr == 'light':
+        payload = {"unique_id": self.format_topic(name, 'illuminance', separator="_"),
```

Comment:
Shouldn't unique_id contain mac at least ? This way it seems easy way to have unique_id conflict, with other stuff in HA.

Reply:
basicly you are right, but i use the name of the sensor as id.
This name is in HA also unique, so there should be no problem.

Reply:
Maybe it should be reflected in README then, that device name should be unique in HA, when using discovery ? I've take a look how esphome is doing it:

```
"unique_id":"ESPswitchsummerhouse_heater_2_switch","device":{"identifiers":"807d3a333ad9","name":"heater_2","sw_version":"esphome v1.12.2 May  5 2019, 22:06:56","model":"PLATFORMIO_ESP01_1M","manufacturer":"espressif"}}
```
Maybe lets append some prefix with bt-mqtt-gateway before format_topic, and maybe worker using ( https://github.com/zewelor/bt-mqtt-gateway/blob/a655504f7833366cd53368ba3d832bec3d642d8d/workers/base.py#L15 ).

Reply:
a changed the id generation and updated the example in the pr 😉